From 0ae2f3441b30922bde8f9fa768c72fa16fe88f45 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Sat, 24 May 2025 15:50:14 +0800
Subject: [PATCH 7/9] WIP: phy: qcom-qmp-combo: set default orientation to
 reverse


diff --git a/drivers/phy/qualcomm/phy-qcom-qmp-combo.c b/drivers/phy/qualcomm/phy-qcom-qmp-combo.c
index 2b77cea809ed..d08fbc158d66 100644
--- a/drivers/phy/qualcomm/phy-qcom-qmp-combo.c
+++ b/drivers/phy/qualcomm/phy-qcom-qmp-combo.c
@@ -1981,6 +1981,13 @@ static bool qmp_combo_configure_dp_mode(struct qmp_combo *qmp)
 	const struct phy_configure_opts_dp *dp_opts = &qmp->dp_opts;
 	u32 val;
 
+	struct device *dev = qmp->dev;
+
+	if (reverse)
+		dev_info(dev, "Configuring DP for reverse orientation\n");
+	else
+		dev_info(dev, "Configuring DP for normal orientation\n");
+
 	val = DP_PHY_PD_CTL_PWRDN | DP_PHY_PD_CTL_AUX_PWRDN |
 	      DP_PHY_PD_CTL_PLL_PWRDN | DP_PHY_PD_CTL_DP_CLAMP_EN;
 
@@ -3365,7 +3372,7 @@ static int qmp_combo_probe(struct platform_device *pdev)
 
 	qmp->dev = dev;
 
-	qmp->orientation = TYPEC_ORIENTATION_NORMAL;
+	qmp->orientation = TYPEC_ORIENTATION_REVERSE;
 
 	qmp->cfg = of_device_get_match_data(dev);
 	if (!qmp->cfg)
-- 
2.51.2

