From acc343ca5b620ee569bc22c008afae16ea349d21 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Fri, 25 Jul 2025 12:49:28 +0800
Subject: [PATCH 4/6] scsi: ufs: qcom: Add support for UFS module detection

Signed-off-by: Xilin Wu <sophon@radxa.com>

Upstream-Status: Inappropriate [Yocto specific]

---
 drivers/ufs/host/ufs-qcom.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)
diff --git a/drivers/ufs/host/ufs-qcom.c b/drivers/ufs/host/ufs-qcom.c
index 1e347092a7d8..fbab9297b9d6 100644
--- a/drivers/ufs/host/ufs-qcom.c
+++ b/drivers/ufs/host/ufs-qcom.c
@@ -2285,6 +2285,20 @@ static int ufs_qcom_probe(struct platform_device *pdev)
 {
 	int err;
 	struct device *dev = &pdev->dev;
+	struct gpio_desc *cd_gpio;
+
+	cd_gpio = devm_gpiod_get_optional(dev, "cd", GPIOD_IN);
+	if (IS_ERR(cd_gpio)) {
+		return dev_err_probe(dev, PTR_ERR(cd_gpio),
+				     "failed to get card-detect GPIO\n");
+	}
+
+	if (cd_gpio) {
+		if (!gpiod_get_value(cd_gpio)) {
+			dev_warn(dev, "UFS module not present, returning -ENODEV\n");
+			return -ENODEV;
+		}
+	}
 
 	/* Perform generic probe */
 	err = ufshcd_pltfrm_init(pdev, &ufs_hba_qcom_vops);
-- 
2.55.0

